internal/platform/web/web.go

```diff
@@ -78,10 +78,7 @@ func (a *App) Handle(verb, path string, handler Handler, mw ...Middleware) {
 		// Set the parent span on the outgoing requests before any other header to
 		// ensure that the trace is ALWAYS added to the request regardless of
 		// any error occuring or not.
-		data, err := trace.MarshalSpanContext(span.SpanContext())
-		if err == nil {
-			w.Header().Set(TraceIDHeader, string(data))
-		}
+		w.Header().Set(TraceIDHeader, v.TraceID)
```

Comment:
This is going to hurt me on the sidecars. Let's talk. I am trying to make sure there is a single picture of trace across process boundaries. I could not find any other way to do it.

Reply:
Look at these lines of code. How do you do this with just a trace id. I couldn't figure it out

https://github.com/ardanlabs/service/blob/master/internal/platform/web/web.go#L64

Reply:
I think you already checking for existing trace on your WithSpanContext https://github.com/ardanlabs/service/blob/master/internal/platform/trace/trace.go#L21

and also you are checking same span.SpanContext which is returned from WithSpanContext and in that case i think value always should be same.

https://github.com/ardanlabs/service/blob/master/internal/platform/web/web.go#L81

Reply:
Or maybe we can create separate header fields for SpinID and TraceOptions if you need to pass it to sidecars

Reply:
When a new span is created in process B, it needs to belong to the same trace that was started in process A. It looks like this concept is missing in the API, which is why I am marshaling that entire value. Doing this I got it to work.

Maybe there is something new in the API that supports this. Or, maybe I am confused and missing something.

Reply:
@farrukhny we should finalize this issue sometime this week.

Reply:
@ardan-bkennedy I am not sure what the issue maybe I am missing something but I did a test of the PR and looks like everything working, please take a look at screenshots

![screen shot 2018-08-13 at 10 29 13 am](https://user-images.githubusercontent.com/6372846/44038548-0584ac88-9ee5-11e8-8f2c-0c92541261d6.png)


![screen shot 2018-08-13 at 10 35 15 am](https://user-images.githubusercontent.com/6372846/44038568-0d55b34e-9ee5-11e8-8b93-d5c18137e067.png)


![screen shot 2018-08-13 at 10 35 36 am](https://user-images.githubusercontent.com/6372846/44038582-11927776-9ee5-11e8-96a4-6ee38d42e698.png)

Reply:
Hey @farrukhny when can we get on a call. I have no doubt I'm doing something wrong. But I think we need to talk.

Reply:
@ardan-bkennedy sure, I will send my number on twitter

Reply:
the changes in trace.go are probably important since you identified we are using a deprecated function now. But after our conversation we don't want to make the other changes. So can you rollback the web.go changes and just keep the trace.go changes. @farrukhny

Reply:
Sure, will do that